lightning/src/ln/channelmanager.rs

```diff
@@ -1819,6 +1819,49 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 		} else { false }
 	}
 
+	// Fail a list of HTLCs. Only called when 1 or more HTLCs fails to be freed
+	// from the holding cell. In this case, the HTLCs need to be failed
+	// backwards or, if they were one of our outgoing HTLCs, then their failure
+	// needs to be surfaced to the user.
+	fn fail_htlcs(&self, mut htlcs_to_fail: Vec<(HTLCSource, PaymentHash)>, channel_id: [u8; 32]) -> Result<(), MsgHandleErrInternal> {
+		for (htlc_src, payment_hash) in htlcs_to_fail.drain(..) {
+			match htlc_src {
```

Comment:
I feel like this could be DRY'd, but didn't dig into it.

Reply:
I DRY'd it a liiittle bit but LMK if you have a better way in mind.